zcbor/zcbor.py

```diff
@@ -1045,20 +1046,32 @@ def skip_condition(self):
         return False
 
     def set_skipped(self, skipped):
+        was_set = self.skipped is not None
         if self.range_check_condition() \
            and self.repeated_single_func_impl_condition() \
            and not self.key:
             self.skipped = True
         else:
             self.skipped = skipped
 
+    def delegate_type_condition(self):
+        """Whether to use the C type of the first child as this type's C type"""
+        ret = self.type in ["LIST", "MAP", "GROUP"]
+        return ret
+
+    def is_delegated_type(self):
+        return self.is_delegated
+
     # Recursively set the access prefix for this element and all its children.
-    def set_access_prefix(self, prefix):
+    def set_access_prefix(self, prefix, is_delegated=False):
         self.accessPrefix = prefix
+        self.is_delegated = is_delegated
         if self.type in ["LIST", "MAP", "GROUP", "UNION"]:
-            list(map(lambda child: child.set_access_prefix(self.var_access()),
+            list(map(lambda child: child.set_skipped(child.skip_condition()),
```

Comment:
So by default every variable is accessible then you are skipping those who you don't want?

Reply:
It's a change in the sense that it fixes a bug :). Note that this is replacing old line 1083.